stream_alert/shared/lookup_tables/core.py

```
@@ -0,0 +1,160 @@
+            driver = construct_persistence_driver(table_configuration)
+            self._tables[table_name] = LookupTable(
+                table_name,
+                driver,
+                table_configuration
+            )
+
+        LOGGER.info('LookupTablesCore initialized!')
+
+    def table(self, table_name):
+        """
+        Drop-in replacement method for table(), on the original LookupTables implementation, with
+        a small caveat; the original implementation returns a dict, whereas this one returns an
+        individual LookupTable instance.
+
+        Params:
+            table_name (str)
+
+        Returns:
+            LookupTable
+        """
+        return self.get_table(table_name)
+
+    def get_table(self, table_name):
+        """
+        Retrieves the instance of the LookupTable requested. If the requested table name is not
+        found, it will Log an error, and subsequently return a NullTable.
+
+        Params:
+            table_name (str): The name of the lookup table desired
+
+        Returns:
+            LookupTable: An individual instance. Not to be mixed up with LookupTables (plural!)
+        """
+        if table_name in self._tables:
+            return self._tables[table_name]
+
+        LOGGER.error(
+            (
+                'Nonexistent LookupTable \'%s\' referenced. Defaulting to null table. '
+                'Valid tables were (%s)'
+            ),
+            table_name,
+            ', '.join(self._tables.keys())
+        )
+        # FIXME (derek.wang) Would it be preferable to raise an exception instead?
+        #  In the current implementation if it returns a NullTable, the code could continue
+        #  executing with None values everywhere, which may cause undesirable behavior. Maybe
+        #  fail fast?
+        return self._null_table
+
+        # raise LookupTablesError('Nonexistent LookupTable {} referenced.'.format(table_name))
```

you could theoretically raise this here, but beware that (relevant to how these are currently used) rule logic is wrapped in a bare except that will just log the exception anyway. logging here is probably fine for now